pkg/kubectl/cmd/get.go

```diff
@@ -326,6 +326,11 @@ func RunGet(f *cmdutil.Factory, out io.Writer, errOut io.Writer, cmd *cobra.Comm
 			allErrs = append(allErrs, err)
 		}
 		printer.AfterPrint(errOut, res)
+
+		// output empty list warning when no resources to display
+		if len(infos) == 0 {
+			fmt.Fprintf(errOut, "%s %q %s\n", "No resources found in namespace", cmdNamespace, "that match your criteria.")
```

Comment:
Get can be used on cluster scoped or across multiple namespaces so this message is too specific.  I'd recommend just using a more generic message "No resources found".  The alternative is a more complex integration into resource.Builder - you'd need to accumulate and store markers of intent when the resource builder crafts a Result and then have the Result be able to spit out a non-generic message.  Not sure it's worth it.